scanrepository/scanrepository.go

```diff
@@ -308,21 +308,18 @@ func (cfp *ScanRepositoryCmd) fixSinglePackageAndCreatePR(vulnDetails *utils.Vul
 	}
 
 	workTreeIsClean, err := cfp.gitManager.IsClean()
+	if err != nil {
+		return
+	}
 	if !workTreeIsClean {
-		var removeTempDirCallback func() error
-		err, removeTempDirCallback = cfp.gitManager.CreateBranchAndCheckoutWithCopyingFilesDiff(fixBranchName)
-		defer func() {
-			err = errors.Join(err, removeTempDirCallback())
-		}()
-		if err != nil {
-			err = fmt.Errorf("failed while creating branch %s: %s", fixBranchName, err.Error())
-			return
-		}
+		// If there are local changes, such as files generated after running an 'install' command, we aim to preserve them in the new branch
+		err = cfp.gitManager.CreateBranchAndCheckoutWithLocalChanges(fixBranchName)
 	} else {
-		if err = cfp.gitManager.CreateBranchAndCheckout(fixBranchName); err != nil {
-			err = fmt.Errorf("failed while creating branch %s: %s", fixBranchName, err.Error())
-			return
-		}
+		err = cfp.gitManager.CreateBranchAndCheckout(fixBranchName)
+	}
```

Comment:
Let's reduce code duplication, these functions have shared logic, I suggest adding a boolean flag and calling one function:

```suggestion
	if err = cfp.gitManager.CreateBranchAndCheckout(fixBranchName, workTreeIsClean); err != nil {
        return err
    }
```